gamification-github-services/src/main/java/org/exoplatform/gamification/github/rest/GithubWebHookRest.java

```diff
@@ -23,153 +23,137 @@
+    this.connectorService = connectorService;
+    this.webhookService = webhookService;
   }
 
   @POST
   @Consumes(MediaType.APPLICATION_JSON)
   @Path("webhooks")
   public Response githubEvent(// NOSONAR
-                              @HeaderParam("x-github-event")
-                              String event,
-                              @HeaderParam("x-hub-signature")
-                              String signature,
+                              @HeaderParam("x-github-event") String event,
+                              @HeaderParam("x-hub-signature") String signature,
                               String obj) {
-    if (Utils.verifySignature(obj, signature, githubHooksManagement.getSecret())) {
-
-      try {
-        ConnectorService connectorService = CommonsUtils.getService(ConnectorService.class);
-        ObjectMapper objectMapper = new ObjectMapper();
-        JsonNode infoNode = objectMapper.readTree(obj);
-        String ruleTitle = "";
-        String senderId = "";
-        String receiverId = "";
-        String object = "";
-        String githubId = "";
-
-        switch (event) {
-          case PUSH_EVENT_NAME: {
-            ruleTitle = "pushCode";
-            githubId = infoNode.get("pusher").get("name").textValue();
-            // TO DO
-            // to move to gamification service after the implementation of the gamification trigger service
-            senderId = connectorService.getAssociatedUsername(CONNECTOR_NAME, githubId);
-            if (senderId != null) {
-              Identity socialIdentity = getUserSocialId(senderId);
-              if (socialIdentity != null) {
-                receiverId = senderId;
-                object = infoNode.get("head_commit").get("url").textValue();
-                githubHooksManagement.broadcastGithubEvent(ruleTitle, senderId, receiverId, object);
-              }
-            }
-          }
-            break;
-          case PULL_REQUEST_EVENT_NAME: {
-            githubId = infoNode.get("sender").get(GITHUB_USERNAME).textValue();
-            senderId = connectorService.getAssociatedUsername(CONNECTOR_NAME, githubId);
-            if (infoNode.get("action").textValue().equals("opened")) {
-              ruleTitle = "creatPullRequest";
-              if (senderId != null) {
-                Identity socialIdentity = getUserSocialId(senderId);
-                if (socialIdentity != null) {
-                  receiverId = senderId;
-                  object = infoNode.get(PULL_REQUEST_EVENT_NAME).get("html_url").textValue();
-                  githubHooksManagement.broadcastGithubEvent(ruleTitle, senderId, receiverId, object);
-                }
-              }
-            }
 
+    if (!webhookService.verifyWebhookSecret(obj, signature)) {
+      return Response.status(Response.Status.UNAUTHORIZED).build();
+    }
+    try {
+      ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
please move this business logic inside a Service layer and split this into sub methods which each shouldn't be more than 15 to 20 lines + refactor to reuse common parts in Utils methods.